apps/synapse-portal-framework/src/components/CollapsibleSection.tsx

```diff
@@ -0,0 +1,136 @@
+    linkDerivedFromEntityId ? id : undefined,
+  )
+
+  if (linkDerivedFromEntityId && isLoading) {
+    return <Skeleton width={300} />
+  }
+
+  return (
+    <Box
+      id={id}
+      onMouseOver={() => {
+        setShowLink(true)
+      }}
+      onMouseOut={() => {
+        setShowLink(false)
+      }}
+    >
+      <Box sx={{ gap: '30px' }}>
+        {!hideTitle && title && (
+          <Box
+            sx={{
+              display: 'flex',
+              borderBottom: '1px solid',
+              borderColor: 'grey.400',
+              marginBottom: '30px',
+            }}
+            onClick={() => setOpen(!open)}
+          >
+            <Typography
+              variant="sectionTitle"
+              sx={{
+                display: 'flex',
+                marginBottom: '14px',
+              }}
+            >
+              {linkDerivedFromEntityId
+                ? `${entityTitlePrepend}${entityHeader?.name ?? ''}`
+                : title}
+              {helpText && (
+                <Box
+                  sx={{
+                    fontSize: '14px',
+                    display: 'inline-block',
+                    ml: '5px',
+                  }}
+                >
+                  <HelpPopover markdownText={helpText} />
+                </Box>
+              )}
+              <span
+                style={{
+                  marginTop: '-3px',
+                  ...(showLink ? { display: 'inline' } : { display: 'none' }),
+                }}
+              >
+                <Tooltip
+                  title={copied ? 'Copied' : 'Copy link to section'}
+                  placement="right"
+                >
+                  <div
+                    style={{ cursor: 'pointer' }}
+                    onClick={e => {
+                      e.stopPropagation()
+                      const urlWithoutHash = window.location.href.replace(
+                        window.location.hash,
+                        '',
+                      )
+                      const url = `${urlWithoutHash}#${id}`
+                      copyStringToClipboard(url).then(() => {
+                        setCopied(true)
+                      })
+                    }}
+                  >
+                    <IconSvg icon="link" wrap={false} sx={{ pl: 1 }} />
+                  </div>
+                </Tooltip>
+              </span>
+            </Typography>
+            {open ? (
+              <KeyboardArrowUpIcon
+                sx={{
+                  marginLeft: 'auto',
+                  cursor: 'pointer',
+                  color: '#878E95',
+                }}
+              />
+            ) : (
+              <KeyboardArrowDownIcon
+                sx={{
+                  marginLeft: 'auto',
+                  cursor: 'pointer',
+                  color: '#878E95',
+                }}
+              />
+            )}
+          </Box>
+        )}
+        <Box sx={{ paddingBottom: open ? '50px' : 0 }}>{open && children}</Box>
```

Comment:
Use the MUI Collapse component to wrap the children, which will handle showing/hiding and give us a gentle animation